networklibrary/src/main/java/com/eis/communication/network/SMSJoinableNetManager.java

```diff
@@ -0,0 +1,35 @@
+package com.eis.communication.network;
+
+import com.eis.communication.network.ListenersInterfaces.JoinInvitationListener;
+import com.eis.smslibrary.SMSPeer;
+
+/**
+ * Concrete JoinableNetwork for SMS Messages
+ *
+ * @author Marco Cognolato
+ */
+public class SMSJoinableNetManager extends SMSNetworkManager
```

Comment:
You're using basic `FailReason` as fail reason list, couldn't be that there are more reasons why a sms network might fail? Did you think about this (adding a `SMSReplicatedNetFailReason` class)? Are you already willing to do this in the future?

Reply:
I'm definitely gonna either change FailReason to a more inclusive one, I just didn't want to add to much stuff to an already pretty big pr.